### ISSUE TITLE
Daily build of phoenicis-master

### ISSUE THREAD

**Issue by t0msk** [PhoenicisOrg/phoenicis#1739]
Hello,

this refers to https://github.com/PhoenicisOrg/phoenicis/issues/1737 it looks like that Ubuntu users cannot build phoenicis from git anymore, so idea is that you could setup script which will build phoenicis-master every day.

**Comment by madoar** [comment]
@qparis as an alternative: how about we provide a script which uses docker to build the `deb` file?

**Comment by qparis** [comment]
Yes why not. 

But I'm wondering if it would not be easier to provide a nightly repository for people that wants to use the snapshot

**Comment by qparis** [comment]
Both can be done I guess

**Comment by madoar** [comment]
Yes. The nightly can be used for people who don't want to do the build process themselves, while the docker script can be used for people who don't want to setup their build environment themselves.

**Comment by ImperatorS79** [comment]
Cannot you use travis to provide the builded .deb as a badge near the travis badge ? (like appveyor for example)

**Comment by Kreyren** [comment]
if you want daily build of master gentoo/funtoo is practically made for this usecase :thinking: Ubuntu is quite the opposite..

**Comment by plata** [comment]
Travis can create GitHub releases. Could also be an option. However, I thing we've talked about this topic already and we came to the conclusion that we do not want it because it will be too easy for people to download nightlies and then start complaining that they are broken.

**Comment by qparis** [comment]
I was more thinking about providing a deb repository

**Comment by plata** [comment]
Where the deb is provided doesn't really change the argument not to provide nightlies.

**Comment by t0msk** [comment]
@plata It is stupid argument, because everybody can clone git repo, build it and complain.. Problem is for Ubuntu users.

**Comment by plata** [comment]
I'm not so sure. If people cannot handle JDK 11, why should they be able to cope with all the crazy issues that might show up in nightlies? If you want to look at devel snapshots, use `mvn exec:java`. The .deb is really not made for this purpose (and it might not even work if you try to install the "same" deb version twice).

**Comment by ImperatorS79** [comment]
If you state the ppa is really unstable (like padoka ppa for mesa) (in the name for example) ?

**Comment by t0msk** [comment]
@plata For example I can install JDK 11, but I don't want to install it manually, I want it from repo, if you install "same" deb it will work, old is removed and new is installed, I tried it.

**Comment by madoar** [comment]
Docker has the huge advantage in my opinion, that it allows for the creation of a non changing build environment, independent on the host it is run on.

**Comment by qparis** [comment]
I think the two proposals can be done. 
The repository is easier and it has the advantage of gathering more people to test phoenicis

**Comment by Kreyren** [comment]
tldr what is the status? i made a system that builds master every X automatically (if buildable) and uploads release locally expecting it to release on kreyrenicis repo, might be usable here?

**Comment by plata** [comment]
Actually, we have the packaging builds in Travis anyways. If we decide to do so, we could upload those somewhere.

**Comment by Kreyren** [comment]
https://media1.tenor.com/images/98933dfc1457d75a894508861f595e98/tenor.gif?itemid=12406023

**Comment by Kreyren** [comment]
@plata 
![](https://media1.tenor.com/images/98933dfc1457d75a894508861f595e98/tenor.gif?itemid=12406023)

**Comment by plata** [comment]
The question is: where? As a GitHub release or on our website or something else? In any case, I cannot decide this on my own.

**Comment by Kreyren** [comment]
@plata i would just release it on github.

**Comment by plata** [comment]
@qparis @madoar shall we do this?

**Comment by Kreyren** [comment]
@plata yes

**Comment by plata** [comment]
What I'm not sure about is if this will create an enormous amount of releases or if one release can be updated all the time. Apart from that, snapshot builds are not really releases strictly speaking.

**Comment by plata** [comment]
Just read the docs again and my understanding is: Every GitHub release (also if created by Travis) needs a tag. Therefore, we would basically tag every commit. I don't think that's a good idea.

@qparis could we build in a docker container every night and publish the latest build on the website directly?

**Comment by Kreyren** [comment]
tag != release afaik.. releases are tagged, but those are not present in branches on github.

**Comment by qparis** [comment]
I would just create an automated deb repository. Assuming that the packager is not broken, this is easy to do. Just a matter of time :-)

**Comment by madoar** [comment]
I would add the "normal" releases (`deb`, `dmg`, `jar`, ...) to GitHub releases (alpha-1, alpha-2 and alpha-3). In addition I agree with @qparis that we could add a nightly build deb repository for all interested people.

**Comment by Kreyren** [comment]
If you provide standalone pre-compiled binarry i can also make downstream compatibility for gentoo and exherbo.

**Comment by plata** [comment]
I'm fine with that approach. So to summarize:
- normal releases like alpha, beta etc. will be on GitHub and include deb, dmg, jar...
- nightly builds will be published for deb only in a PPA
- if nightly builds are required for other distributions, their maintainers are free to provide packages (e.g. AUR)

**Comment by Kreyren** [comment]
@plata assuming that i can extract standalone jar from it -> perfect.
